cli/js/tests/url_test.ts

```diff
@@ -132,6 +132,43 @@ unitTest(function urlSearchParamsReuse(): void {
   assert(sp === url.searchParams, "Search params should be reused.");
 });
 
+unitTest(function urlBackSlashes(): void {
+  const url = new URL(
+    "https:\\\\foo:bar@baz.qat:8000\\qux\\quux?foo=bar&baz=12#qat"
+  );
+  assertEquals(
+    url.href,
+    "https://foo:bar@baz.qat:8000/qux/quux?foo=bar&baz=12#qat"
+  );
+});
+
+unitTest(function urlRequireHost(): void {
+  assertEquals(new URL("file:///").href, "file:///");
+  assertThrows(() => {
+    new URL("ftp:///");
+  });
+  assertThrows(() => {
+    new URL("http:///");
+  });
+  assertThrows(() => {
+    new URL("https:///");
+  });
+  assertThrows(() => {
+    new URL("ws:///");
+  });
+  assertThrows(() => {
+    new URL("wss:///");
+  });
+});
+
+unitTest(function urlDriveLetter() {
+  assertEquals(
+    new URL("file:///C:").href,
+    Deno.build.os == "windows" ? "file:///C:/" : "file:///C:"
```

Comment:
This seems wrong. URL is os dependent?

Reply:
This is not what the browser returns on windows.

Reply:
Yeah. I don't know if it's in the specification but Chrome supports it. It's particularly useful for Deno/Windows users. A file URL is supposed to specifically contain a native FS path so it seems only correct.

Reply:
URL needs to match the browser exactly. So maybe we can add some other utility to handle this?

Reply:
> This is not what the browser returns on windows.

![7](https://user-images.githubusercontent.com/29990554/80991297-b1740800-8e2f-11ea-9fe8-ec0c7c7d69a6.png)

Chrome Version 81.0.4044.129.

Reply:
![8](https://user-images.githubusercontent.com/29990554/80992662-c2be1400-8e31-11ea-9684-ff0b857411a0.png)

Firefox Version 72.0.2